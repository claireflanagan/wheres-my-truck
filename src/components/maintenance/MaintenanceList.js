import React, { Component, Fragment } from 'react';
import styles from './MaintenanceList.css';

export default class MaintenancesList extends Component {

  render() {
    const { maintenances } = this.props;
    console.log('maintenances', maintenances);
    const tableRows = maintenances.maintenances.map((maintenance, i) => {
      return (
        <tr key={i}>
          <td>{maintenance.date}</td>
          <td>{maintenance.type}</td>
          <td>{maintenance.notes}</td>
        </tr>
      );
    });
    return (
      <Fragment>  
        <h1>{maintenances.details.name}</h1> 
        <table className={styles.maintTable}>
          <thead>
            <th>Date</th>
            <th>Type</th>
            <th>Notes</th>
          </thead>
          <tbody>
            {tableRows}
          </tbody>          
        </table>
      </Fragment>
     
    );
  }

}
