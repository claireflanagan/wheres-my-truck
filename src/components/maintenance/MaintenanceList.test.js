import React from 'react';
import { shallow } from 'enzyme';
import MaintenanceList from './MaintenanceList';

describe('MaintenanceList component', () => {
  const maintenances = {
    details: {
      id: 'a0',
      name: 'Truck 0',
      location: '10th and Couch',
      vin: '1FTFS24Y8PHA25143',
      plates: '876ABC',
      year: 2018,
      make: 'Truck',
      model: 'Big Truck',
      tireSize: 20,
      boughtDate: '2017-09-10',
      registration: 'https://www.dmv.ca.gov/imageserver/dmv/images/vr/regcard_w_arrow2.jpg',
      insurance: 'https://approvedauto.files.wordpress.com/2013/12/id-card-example.jpg'
    },
    maintenances: [
      {
        type: 'oil change',
        date: '2018-10-31',
        notes: 'this was hard!'
      }
    ]
  };
  it('matches a snapshot', () => {
    expect(true);
    const wrapper = shallow(<MaintenanceList
      maintenances={maintenances}/>);
    expect(wrapper).toMatchSnapshot();
        
  });
});
