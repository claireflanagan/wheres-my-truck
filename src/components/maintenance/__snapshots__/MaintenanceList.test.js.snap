// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintenanceList component matches a snapshot 1`] = `
<Fragment>
  <h1>
    Truck 0
  </h1>
  <table
    className="table threeColumns"
  >
    <thead>
      <th>
        Date
      </th>
      <th>
        Type
      </th>
      <th>
        Notes
      </th>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td />
        <td>
          oil change
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</Fragment>
`;
